File: client/src/app/_models/Vehicle.ts
import { Client } from "./Client";
import { Order } from "./Order";

export interface Vehicle {
    id: number;
    brand: string;
    model: string;
    registrationNumber: string;
    vin: string;
    engineCode?: string;
    capacity: number;
    engineFuel: string;
    enginePower: number;
    enginePowerUnit: string;
    color: string;
    productionDate: Date;
    firstRegistration: Date;
    technicalInspectionEnd?: Date;
    description?: string;
    currentOwner: Client;
    createdAt: Date;
    updatedAt: Date;
    orders: Array<Order>;
}

File: API/Entities/Vehicle.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities
{
    public class Vehicle
    {
        public int Id { get; set; }
        [Required]
        public string Brand { get; set; }
        [Required]
        public string Model { get; set; }
        [Required]
        public string RegistrationNumber { get; set; }
        public string Vin { get; set; }
        public string EngineCode { get; set; }
        public int? Capacity { get; set; }
        public string EngineFuel { get; set; }
        public int? EnginePower { get; set; }
        public string EnginePowerUnit { get; set; }
        public string Color { get; set; }
        public int? ProductionDate { get; set; }
        public DateTime? FirstRegistration { get; set; }
        public DateTime? TechnicalInspectionEnd { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Foreign keys
        public int AppUserId { get; set; }
        public int? CurrentOwnerId { get; set; }
        public AppUser AppUser { get; set; }
        public Client CurrentOwner { get; set; }
        public ICollection<Order> Orders { get; set; }
    }
}